File: tests/measurement_utils_test.cpp
#include "jetbus/measurement_utils.hpp"

#include <catch2/catch_approx.hpp>
#include <catch2/catch_test_macros.hpp>

TEST_CASE("digit_to_double and double_to_digit roundtrip", "[measurement]") {
    const int decimals = 3;
    const std::int32_t digits = 12345;
    const double value = jetbus::digit_to_double(digits, decimals);
    REQUIRE(value == Catch::Approx(12.345));
    const auto restored = jetbus::double_to_digit(value, decimals);
    REQUIRE(restored == digits);
}

TEST_CASE("string_to_bool interprets zero as false", "[measurement]") {
    REQUIRE_FALSE(jetbus::string_to_bool("0"));
    REQUIRE(jetbus::string_to_bool("1"));
    REQUIRE(jetbus::string_to_bool("42"));
}

File: tests/measurementUtils.test.ts
import { describe, expect, it } from "vitest";
import { digitToDouble, doubleToDigit, stringToBool } from "../src/jetbus/measurementUtils";

describe("measurementUtils", () => {
  it("converts digits to doubles respecting decimals", () => {
    expect(digitToDouble(12345, 2)).toBeCloseTo(123.45, 6);
    expect(digitToDouble(-9876, 3)).toBeCloseTo(-9.876, 6);
  });

  it("converts doubles to digits respecting decimals", () => {
    expect(doubleToDigit(123.45, 2)).toBe(12345);
    expect(doubleToDigit(-9.876, 3)).toBe(-9876);
  });

  it("parses string to boolean", () => {
    expect(stringToBool("0")).toBe(false);
    expect(stringToBool("1")).toBe(true);
    expect(stringToBool("anything")).toBe(true);
  });
});
